indico/modules/events/registration/controllers/management/invitations.py

```diff
@@ -105,6 +105,20 @@ def _checkParams(self, params):
         RHManageRegFormBase._checkParams(self, params)
         self.invitation = RegistrationInvitation.get_one(request.view_args['invitation_id'])
 
+
+class RHRegistrationFormDeleteInvitation(RHRegistrationFormInvitationBase):
+    """Delete a registration invitation"""
+
     def _process(self):
         db.session.delete(self.invitation)
         return jsonify_data(invitation_list=_render_invitation_list(self.regform))
+
+
+class RHRegistrationFormMarkInvitationAsDeclined(RHRegistrationFormInvitationBase):
+    """Mark a registration is declined by the invitee."""
+
+    def _process(self):
+        if self.invitation.state == InvitationState.pending:
+            self.invitation.state = InvitationState.declined
+            db.session.flush()
```

Comment:
Yes, we have autoflush (so any SELECT is going to flush) and also always commit after a successful request (which also triggers a flush). Nothing wrong with doing an explicit flush though, it has the advantage of getting errors (if something breaks) closer to the code that actually caused them.